parlai/core/torch_agent.py

```diff
@@ -1101,6 +1116,14 @@ def share(self):
         Subclasses will likely want to share their model as well.
         """
         shared = super().share()
+
+        if self.opt.get('numthreads', 1) > 1 and isinstance(self.metrics, dict):
+            torch.set_num_threads(1)
```

Comment:
what does this call do? looks like we weren't calling this in TGA before if `self.opt['numthreads'] > 1`

Reply:
It prevents torch from parallelizing the matrix multiplication across cores (resulting in hogwild stomping each other).

We actually already do it in `__init__` so I can drop this line.